idaes/apps/matopt/util/util.py

```diff
@@ -51,7 +51,8 @@ def myArrayEq(x, y, atol):
     Returns:
     (bool) true if the two arrays are equal
     """
-    return (np.abs(x - y) < atol).all()
+    return (x[0] - y[0] < atol) and (x[0] - y[0] > -atol) and (x[1] - y[1] < atol) and (x[1] - y[1] > -atol) and (
+            x[2] - y[2] < atol) and (x[2] - y[2] > -atol)
```

Comment:
I'm curious why you went with expanding the comparison operation this way. You say it's for speed-up, is that because the numpy abs() function call would take longer than this chain of ANDs?

Reply:
Yes, this expansion makes the canvas construction significantly faster in my testing. I think that's because the previous code needs to calculate the difference vector first and then absolute values for all coordinates. This expansion essentially decomposes all calculations into small chunks. When one of the conditions is not met, it will return false without calculating more coordinates. I am using indexed access, assuming we are dealing with 3d points, which should always be valid for materials design. Besides, many assertions in the canvas construction process that makes sure the points are 3d.